OFEP-flag-eval-wire-protocol.md

```diff
@@ -0,0 +1,55 @@
+## Flag Evaluation Wire Protocol
+
+## State: DRAFTING
+
+Create a wire protocol for RPC-based flag evaluation compliant with the OpenFeature specification.
+
+## Background
+
+OpenFeature offers a vendor-agnostic SDK that works across various open source and commercial offerings.
+Developers can choose the provider they want to use and include it in their codebase.
+This works well when a high-quality provider exists, and it's acceptable to include a vendor-specific dependency.
+However, that's not always the case.
+For example, many open source applications could benefit from having a standardized means of including feature flags.
+A flag evaluation wire protocol would allow developers to choose the architecture that best fits their needs.
+
+## In-process vs Remote Evaluation
+
+Many feature flag vendors offer SDKs that perform flag evaluation in-process.
+This provides fast and consistent flag evaluations with no additional deployment complexities.
+Remote evaluation logic can be written in a single language and be used interchangeably with any system complying with the wire protocol.
+That makes it ideal for open source projects that need a vendor-agnostic solution for feature flags.
```

Comment:
> That makes it ideal for open source projects that need a vendor-agnostic solution for feature flags.

This doesn't follow to me. A big part of the OpenFeature value prop is in providing a vendor-neutral (in-process) SDK, one which an open-source project can plug into without any awareness of what providers are plugged into that SDK.

Reply:
The problem that @oleg-nenashev reported with the current approach, is that it is hard for opensource projects to support all providers out of the box or to provide a plugin system for people to use a specific provider.

In some languages, we can do it easily (such as Java), but for some others such as Go it can be complex and will force these open-source projects to create a plugin system specific to open-feature to be able to support an unknown list of providers.